### DISCUSSION TITLE
Search for custom - what is wrong?

### DISCUSSION THREAD

**Discussion by Jimmi08** [e107inc/e107#5198] [Q&A Core]
Hi,
I need help. What is wrong?

My searching (e_search is used) and I know that it is because of relevance value.  I simplified query and tested it PHPmyadmin (so no influence of e107 code)

Query for custom plugin
`SELECT SQL_CALC_FOUND_ROWS n.property_id, n.property_title, (MATCH(n.property_title) AGAINST ('Dom' IN BOOLEAN MODE)) AS relevance FROM e107_property as n;`

Result: (see - relevance is zero)

![image](https://github.com/e107inc/e107/assets/5429548/5d16b64d-4dcf-4477-b74a-51b7f169a5a0)

I tried the same thing for the news and it works
 
![image](https://github.com/e107inc/e107/assets/5429548/a95941af-ac9c-43f7-9358-25e65ece0610)

Any idea what can be wrong? Thanks

**Comment by tgtje** [comment]
@Jimmi08 _**wild going here**_, you mentioned e_search, so assuming a file is used in the property plugin? (name guess).
iff so : as news is used as example or alike : looking at news e_search... do you have 'return_fields'	>...  AND 'search_fields' > ...
that can be 'compared' ?  eg return and search both contain    n.property_id, n.property_title, ?

**Reply by Jimmi08** [reply]
@tgtje Yes, it is plugin e_search and it returns nothing because relevance value is 0 - so I used that query directly in phpmyadmin and it returns really zero.  In other database it returns 1 for news.  
Look at the picture - word "Dom" is in the property title, and relevance should be 1 but it is not (and e_search is not working then)

I excluded e107 environment to be sure that it is not related to latest database charset changes. 

Why the query on first image returns zero?  Do you see some mistypo? Why MATCH AGAINST in boolean mode can fail this way?

**Reply by tgtje** [reply]
afaik differences in innodb and mysql table routine happen . all looks ok for both.
hmm search class uses vars like stopwords eg > do  (no idea of function that is, only idea.. )
maybe you need a + sign before Dom    so  '+Dom'   to exactly match
no pain no gain... i am not good in sql at all
 (do you see any difference in (not using code), but do a basic/standard phpadmin search in dbase itself iff that does differ..

**Comment by Jimmi08** [comment]
It is like 3 letter word issue - but I can't see any warning, or stop searching if you use too short word

**Reply by tgtje** [reply]
_Just as reminder_ : database contains 5 arrays, asking myself : does this plugin need an array too? is All Areas search including it?
